mesonpy/__init__.py

```diff
@@ -754,25 +754,11 @@ def __init__(  # noqa: C901
             [binaries]
             python = '{sys.executable}'
         ''')
-        native_file_mismatch = (
-            not self._meson_native_file.exists()
-            or self._meson_native_file.read_text() != native_file_data
-        )
-        if native_file_mismatch:
-            try:
-                self._meson_native_file.write_text(native_file_data)
-            except OSError:
-                # if there are permission errors or something else in the source
-                # directory, put the native file in the working directory instead
-                # (this won't survive multiple calls -- Meson will have to be reconfigured)
-                self._meson_native_file = self._working_dir / '.mesonpy-native-file.ini'
-                self._meson_native_file.write_text(native_file_data)
-
-        # Don't reconfigure if build directory doesn't have meson-private/coredata.data
-        # (means something went wrong)
-        # See https://github.com/mesonbuild/meson-python/pull/257#discussion_r1067385517
```

Comment:
Can you keep the comment linking to the Github comment?

Reply:
I would prefer to paraphrase Eli's comment in the code comment so it is not necessary to click through the link for the explanation (and we don't rely on the link working in perpetuity). Would that be ok with you?

Reply:
That works for me.